packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/stack-synthesizer.ts

```diff
@@ -22,10 +32,31 @@ export class AuthStackSynthesizer extends LegacyStackSynthesizer {
       const templateName = stack.node.id;
       this.setStackAsset(templateName, template);
     } else {
-      throw new Error(
-        'Error synthesizing the template. Expected Stack to be either instance of AmplifyAuthCognitoStack or AmplifyUserPoolGroupStack',
-      );
+      throw new AmplifyFault('UnknownFault', {
+        message: 'Error synthesizing the template. Expected Stack to be either instance of AmplifyAuthCognitoStack or AmplifyUserPoolGroupStack',
+      });
+    }
+  }
+
+  protected synthesizeTemplate(session: ISynthesisSession, __?: string): FileAssetSource {
```

Comment:
is this implementing an interface? do you have a doc link or context

Reply:
Yes. I linked related change in PR description.
This is protected API from `StackSynthesizer`. They renamed it here https://github.com/aws/aws-cdk/commit/8b2b38187b709a4e9a37a4de043a84267a9ec937#diff-071bf4af4f62cf68b345e393cdb18d584e85f49eea97b34a5319f57dfa64e22a and started calling to new version (which is breaking change btw...).
Both of our synthesizers implement `protected synthesizeStackTemplate(stack: Stack, session: ISynthesisSession): void` . This PR adds the same logic under recently added `protected synthesizeTemplate(session: ISynthesisSession, lookupRoleArn?: string): FileAssetSource`